### DISCUSSION TITLE
v2.2.3 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#3090] [Releases]
**Note: this beta build has breaking changes. Please read the notes below for more information.**

## Installation
Visit [https://getutm.app/install/][1] for the most up to date installation instructions.

## New Features (v2.2.0)
* (macOS 11.3+ Only) **Bridged networking and shared networking support**. You can configure it in the Network settings for your VM.
* (macOS 11+ and iOS 13+) **GPU acceleration for OpenGL on Linux**. Use `virtio-ramfb` or `virtio-gpu-gl-pci` display device and compatible Linux drivers (most modern Linux distros will have it already installed). Windows is not supported because there is currently no virtio-gpu driver for Windows that supports 3D acceleration.

## Changes (v2.2.2)
* (iOS) Enable OpenGL acceleration for iOS 11-12
* (iOS 15+) Increased memory limit for supported devices (M1 iPads currently)
* (macOS) Fix layout issues in settings on macOS Monterey (#2644)
* (macOS) Fix capturing of hotkeys (such as Cmd+Tab) (#2677)
* Disable port forwarding options for shared & bridged networking
* Fixed crash when PPC screamer is selected on iOS (currently this is still not supported)
* Refactored non-OpenGL rendering code, fixing some random crashes seen in TestFlight reports.

## Changes (v2.2.3)
* (iOS) When a mouse/trackpad is already connected when UTM starts up, it will now be properly detected (#3013)
* (iOS) Handle touch and Apple Pencil input while the mouse is captured (#2669)
* Windows BSOD on boot/setup due to `PAGE_FAULT_IN_NONPAGED_AREA` (#2721)
* EFI Boot is now a configurable option: by default new VMs created for pc, q35, and virt* machines will have EFI enabled. In previous versions, EFI is only enabled for virt* machines. Due to compatibility with boot, existing pc and q35 VMs will NOT have EFI enabled and must be manually turned on in Settings -> System -> Advanced Configuration -> UEFI Boot. This may also require you to re-install the bootloader on your VM. As part of this change, EFI variables will also be properly handled (on both ARM and x86 VMs). Note that if you've configured a custom pflash device for your VM, the new UEFI Boot option will take no effect (same as before).
* Added `virtio-ramfb-gl` device to support GPU Acceleration. On previous beta builds, GL is enabled automatically for `virtio-ramfb`. Now, that display device will **not** get GPU acceleration. This is to be more consistent with the other `-gl` display devices. By default, new VMs created will have `virtio-ramfb-gl` selected for `virt` devices (ARM) and `virtio-vga-gl` selected for `q35` and `pc` devices (x86). If you wish to use GPU acceleration on an existing VM, you need to change your display device to one of these two recommended devices (or any device with `-gl` in the name).

Please check the full list on [Github][2] and help report any bug you find that is not listed.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v2.2.3'>v2.2.3 (Beta)</a>.</em>

**Comment by ktprograms** [comment]
Getting this warning / error on opening:
<img width="328" alt="image" src="https://user-images.githubusercontent.com/71804605/132428469-04ebd8bd-bdc8-4da5-b2a7-ae529f4a0613.png">

Also looking at the GH Actions logs the notarization stapling failed?

**Reply by osy** [reply]
Ugh alright

**Reply by osy** [reply]
It's been reuploaded.

**Reply by ktprograms** [reply]
What was the problem?

**Reply by osy** [reply]
Apple’s servers sometimes lag so when you try to staple immediately after seeing the notarization is successful it may still fail. This just happens sometimes.

**Comment by muratdk23** [comment]
Is gpu acceleration working on windows?

**Reply by lilvadim** [reply]
I guess no, because there's no graphics drivers for windows for virtualization in qemu yet.
